packages/frontend/src/hooks/contracts/useController.ts

```diff
@@ -321,6 +338,78 @@ export const useController = () => {
     return emptyState
   }
 
+  /**
+   * Liquidation price is calculated using this document: https://docs.google.com/document/d/1MzuPADIZqLm3aQu-Ri2Iyk9ZUvDA1D6oOikKwwjSC2M/edit
+   *
+   * If you have any doubts please ask Joe Clark aka alpinechicken 🦔
+   */
+  const calculateLiquidationPriceForLP = (ethCollat: BigNumber, shortAmount: BigNumber, position: Position) => {
+    const liquidity = toTokenAmount(position.liquidity.toString(), 18)
+
+    const ETH_LOWER_BOUND = 500
+    const ETH_UPPER_BOUND = 30000
+
+    const pa = !isWethToken0
+      ? new BigNumber(position?.token0PriceLower.toSignificant(18) || 0)
+      : new BigNumber(1).div(position?.token0PriceUpper.toSignificant(18) || 0)
+    const pb = !isWethToken0
+      ? new BigNumber(position?.token0PriceUpper.toSignificant(18) || 0)
+      : new BigNumber(1).div(position?.token0PriceLower.toSignificant(18) || 0)
+
+    const maxEth = liquidity.times(pb.sqrt().minus(pa.sqrt()))
+    const maxSqth = liquidity.times(new BigNumber(1).div(pa.sqrt()).minus(new BigNumber(1).div(pb.sqrt())))
+
+    const divider = shortAmount.times(1.5).times(normFactor)
+
+    const ethValueFunction = (ethPrice: string) => {
+      const _ethPrice = new BigNumber(ethPrice)
+      const p = _ethPrice
+        .times(normFactor)
+        .times(Math.exp(impliedVol * impliedVol * 0.04794520548))
+        .div(INDEX_SCALE)
+
+      if (p.lt(pa)) {
+        return maxSqth.times(p)
+      }
+      if (p.gt(pb)) {
+        return maxEth
+      }
+
+      return liquidity.times(p.sqrt().times(2).minus(pa.sqrt()).minus(p.div(pb.sqrt())))
+    }
+
+    const fzeroFunction = (ethPrice: string) => {
+      const _result = new BigNumber(ethPrice)
+        .minus(ethValueFunction(ethPrice).plus(ethCollat).times(INDEX_SCALE).div(divider))
+        .toString()
```

Comment:
Both will give the same result in this case. It's not similar to our Arithmetic precedence. BigNumber works based on the order we give, because they are just functions!